### DISCUSSION TITLE
Trigger tag on WooCommerce affiliate link click

### DISCUSSION THREAD

**Discussion by becausebirds** [scale8/scale8-tag-manager-and-analytics#121] [Q&A]
Hello, I've spun up a self-hosted Scale8 install. I'm trying to fire a tag when a "cart" button is clicked on one of my affiliate products on my website.

The button has a class of "single_add_to_cart_button" and I created a tag in Scale8 with a Click Elements event rule of "custom" with the selector : ".single_add_to_cart_button"

This tag fires custom HTML with Plausible Analytics "Custom Event" javascript.

I've deployed the revision and added the global code and the tag to all the product pages, but it doesn't appear to be firing properly, as I don't see any recorded events in Plausible.

I've used GTM previously to fire the needed code on a Click Text trigger, and I'm trying to replace this.

Example URL: [https://becausebirds.com/product/charley-harper-bird-drinking-glasses-2/](url)

Any advice or guidance is much appreciated! Thanks!

**Comment by cwbeck** [comment]
Hi @becausebirds, are you able to share screenshots of the platform along with scripts you are trying to fire?

**Reply by becausebirds** [reply]
Currently deployed in the production environment. I've verified the global code and the tag code appears in the source code of the desired pages. 

Tag rule with custom class selector
.single_add_to_cart_button

![Tag Rule with custom class selector](https://user-images.githubusercontent.com/110354170/183259736-d5e65b75-b0c0-4e3b-b766-bc174d2adea2.png)

Tag script using custom HTML

`<script>
				plausible('Affiliate Click', {props: {product: document.title}});
</script>`
![Tag script with custom html](https://user-images.githubusercontent.com/110354170/183259738-8c97b4a6-7ae6-4b1b-b276-3b211c523364.png)

Example button details being targeted

![Button click class being targeted](https://user-images.githubusercontent.com/110354170/183259739-714f20f1-c3ac-4039-8c57-a534bde3b937.png)

[https://becausebirds.com/product/charley-harper-bird-drinking-glasses-2/](https://www.becausebirds.com/product/charley-harper-bird-drinking-glasses-2/)

Thanks!

**Reply by cwbeck** [reply]
So I believe the problem is scope.

```js
window.top.plausible(....)
```

The HTML you've provided is loaded into an iframe, but that iframe will have its own scope and plausible will not exist in that new scope. Instead you need to look back up.

https://developer.mozilla.org/en-US/docs/Web/API/Window/top

A quick test locally trying to mock your setup and it works for me.
